### PR TITLE
[NO-JIRA] Fix package-lock

### PR DESCRIPTION
**Pull request by tiagoengel**
This only changes package-lock*s* that have changed during the install phase.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-react-native/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
     "android": "npm run android:emulator & npm run android:build",
     "spellcheck": "mdspell -r --en-gb --ignore-acronyms --ignore-numbers --no-suggestions '*.md' '**/*.md' '!**/node_modules/**/*.md' '!ios/Pods/**/*.md'",
     "spellcheck:interactive": "mdspell --en-gb --ignore-acronyms --ignore-numbers '*.md' '**/*.md' '!**/node_modules/**/*.md' '!ios/Pods/**/*.md'",
-    "start": "react-native start"
+    "start": "react-native start",
+    "fix:package-lock": "git status -s | grep package-lock.json | awk '{print $2}' | xargs sed -i 's/http:\\/\\//https:\\/\\//g'"
```
Comment (georgegillams) [inline package.json]: The `-i` expects input

```suggestion
    "fix:package-lock": "git status -s | grep package-lock.json | awk '{print $2}' | xargs sed -i '' 's/http:\\/\\//https:\\/\\//g'"
```

Comment (georgegillams) [reply package.json]: At the moment it is interpretting the regex as the input for `-i` and doing this

Screenshot 2019-03-15 at 15 45 34

Comment (tiagoengel) [reply package.json]: Are you sure? `-i` as far as I know is `in-place`, meaning to replace the file

Comment (tiagoengel) [reply package.json]: Ah, nvm this is that problem with mac os using an old version of sed

Comment (georgegillams) [reply package.json]:
```
    -i extension
             Edit files in-place, saving backups with the specified extension.  If a zero-length extension is given, no backup will be saved.  It is not recommended to give a zero-length extension when in-place editing files, as you risk corruption or partial content in situations where disk space is exhausted, etc.
```

Comment (tiagoengel) [reply package.json]: Yeah, but that is not the behaviour with GNU seed (which I have locally and is by default on linux) and they are not compatible

Comment (tiagoengel) [reply package.json]: So I've replaced this with `replace` (no pun intended) and it works now. I had to remove from post install though as it is not working. I suspect the package lock is only saved on the disk after the post install script runs

File: package.json [hunk 2]
```diff
@@ -45,7 +45,8 @@
     "android": "npm run android:emulator & npm run android:build",
     "spellcheck": "mdspell -r --en-gb --ignore-acronyms --ignore-numbers --no-suggestions '*.md' '**/*.md' '!**/node_modules/**/*.md' '!ios/Pods/**/*.md'",
     "spellcheck:interactive": "mdspell --en-gb --ignore-acronyms --ignore-numbers '*.md' '**/*.md' '!**/node_modules/**/*.md' '!ios/Pods/**/*.md'",
-    "start": "react-native start"
+    "start": "react-native start",
+    "fix:package-lock": "git status -s | grep package-lock.json | awk '{print $2}' | xargs replace --silent 'http://' 'https://'"
```
Comment (georgegillams) [inline package.json]: We should make this run this in `postinstall` - but that won't work on CI as Travis doesn't actually run within the git repo.
So we would need something like this to make Travis happy with running as a postinstall. This command is becomming an absolute unit 😆

```suggestion
    "fix:package-lock": "(git rev-parse --is-inside-work-tree && git status -s | grep package-lock.json | awk '{print $2}' | xargs replace --silent 'http://' 'https://') || true",
```

Comment (georgegillams) [reply package.json]: Alternatively we make it a pre-commit hook, or we just skip the git check, so then we don't need to worry about CI 🤷‍♂️

**Comment by shaundon** [comment]
This is really really old, can we close it?

**Comment by olliecurtis** [comment]
I think we can close this as its been open for a very long time and no longer seems to be an issue

**Pull request closed without merging**